pkg/build/controller/strategy/util.go

```diff
@@ -99,16 +99,31 @@ func setupDockerSecrets(pod *kapi.Pod, pushSecret string) {
 	}...)
 }
 
-// mergeEnvWithoutDuplicates merges two environment lists without having
-// duplicate items in the output list.
-func mergeEnvWithoutDuplicates(source []kapi.EnvVar, output *[]kapi.EnvVar) {
+// mergeTrustedEnvWithoutDuplicates merges two environment lists without having
+// duplicate items in the output list.  Only trusted environment variables
+// will be merged.
+func mergeTrustedEnvWithoutDuplicates(source []kapi.EnvVar, output *[]kapi.EnvVar) {
+
+	// filter out all environment variables except trusted/well known
+	// values, because we do not want random environment variables being
+	// fed into the privileged STI container via the BuildConfig definition.
+	filteredSource := []kapi.EnvVar{}
+	for _, env := range source {
+		switch env.Name {
+		case "BUILD_LOGLEVEL":
```

Comment:
can you make this a global list variable?

var blacklistEnvVarNames=string[]{"BUILD_LOGLEVEL"}